clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1053,6 +1121,54 @@ static target getAccessTarget(QualType FieldTy,
       AccTy->getTemplateArgs()[3].getAsIntegral().getExtValue());
 }
 
+static bool IsFreeFunction(Sema &SemaRef, const FunctionDecl *FD) {
+  if (FD->hasAttr<SYCLAddIRAttributesFunctionAttr>()) {
+    const auto &Attrs = FD->getAttrs();
+    for (unsigned i = 0, e = Attrs.size(); i < e; i++) {
+      const SYCLAddIRAttributesFunctionAttr *AnAttr =
+          dyn_cast<SYCLAddIRAttributesFunctionAttr>(Attrs[i]);
+      if (AnAttr) {
+        SmallVector<std::pair<std::string, std::string>, 4> NameValuePairs =
+            AnAttr->getAttributeNameValuePairs(SemaRef.Context);
+        for (const auto &NameValuePair : NameValuePairs) {
+          if (NameValuePair.first == "sycl-range-kernel" ||
+              NameValuePair.first == "sycl-nd-range-kernel" ||
+              NameValuePair.first == "sycl-single-task-kernel")
+            return true;
+        }
+      }
+    }
+  }
+  return false;
+}
+
+static std::string constructFFKernelName(ASTContext &Ctx,
+                                         const FunctionDecl *FD) {
+  IdentifierInfo *Id = FD->getIdentifier();
+  std::string NewIdent = (Twine("__free_function_") + Id->getName()).str();
+  return NewIdent;
+}
+
+// Gets a name for the free function kernel function. The suffix allows a normal
+// device function to coexist with the kernel function.
+static std::pair<std::string, std::string>
+constructFreeFunctionKernelName(Sema &S, const FunctionDecl *KernelCallerFunc,
```

Comment:
What does `KernelCallerFunc` refer to here? This is just the free function right? If so, please rename this variable accordingly.

Reply:
Changed the name.